src/BaseFirestoreRepository.ts

```diff
@@ -89,9 +89,11 @@ export default class BaseFirestoreRepository<T extends IEntity>
     await this.firestoreColRef.doc(id).delete();
   }
 
-  runTransaction(executor: (tran: TransactionRepository<T>) => Promise<void>) {
-    return this.firestoreColRef.firestore.runTransaction(t => {
-      return executor(
+  async runTransaction(
+    executor: (tran: TransactionRepository<T>) => Promise<void>
+  ) {
+    return await this.firestoreColRef.firestore.runTransaction(async t => {
+      return await executor(
```

Comment:
These two awaits are unnecessary

Reply:
same reason as above. mostly ease of grok and stack trace combing.

Reply:
See [here](https://github.com/wovalle/fireorm/pull/81#discussion_r332040901)